src/workshed.ts

```diff
@@ -0,0 +1,390 @@
+    setTrainsetConfiguration(pieces);
+  }
+);
+
+const CMC = new GarboWorkshed(
+  $item`cold medicine cabinet`,
+  () => {
+    return get("_coldMedicineConsults") >= 5;
+  },
+  () => {
+    if (property.getNumber("_nextColdMedicineConsult") > totalTurnsPlayed()) return;
+    const options = visitUrl("campground.php?action=workshed");
+    let i = 0;
+    let match;
+    const regexp = /descitem\((\d+)\)/g;
+    const itemChoices = new Map<Item, number>();
+    if (!globalOptions.noBarf) {
+      // if spending turns at barf, we probably will be able to get an extro so always consider it
+      itemChoices.set($item`Extrovermectin™`, -1);
+    }
+
+    while ((match = regexp.exec(options)) !== null) {
+      i++;
+      const item = descToItem(match[1]);
+      itemChoices.set(item, i);
+    }
+
+    const bestItem = maxBy([...itemChoices.keys()], garboValue);
+    const bestChoice = itemChoices.get(bestItem);
+    if (bestChoice && bestChoice > 0) {
+      visitUrl("campground.php?action=workshed");
+      runChoice(bestChoice);
+    }
+    if (handlingChoice()) visitUrl("main.php");
+  }
+);
+
+const pizzaCube = new GarboWorkshed(
+  $item`diabolic pizza cube`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const asdonMartin = new GarboWorkshed(
+  $item`Asdon Martin keyfob`,
+  () => {
+    return haveEffect($effect`Driving Observantly`) >= estimatedTurns();
+  },
+  () => {
+    const currentTurns = haveEffect($effect`Driving Observantly`);
+    const requiredTurns = estimatedTurns() - currentTurns;
+    const currentFuel = getFuel();
+    const requiredFuel = Math.max(Math.ceil(requiredTurns / 30) * 37 - currentFuel, 0);
+    if (requiredFuel > 10000) {
+      throw new Error(`Wanted to fill ${requiredFuel} fuel in our Asdon, probably a bug!`);
+    }
+    fillTo(requiredFuel);
+    for (let x = 0; x < Math.ceil(requiredTurns / 30); x++) {
+      cliExecute("asdonmartin drive observantly");
+    }
+  }
+);
+
+const mayoClinic = new GarboWorkshed(
+  $item`portable Mayo Clinic`,
+  () => {
+    return dietCompleted;
+  },
+  () => {
+    return;
+  }
+);
+
+const DNALab = new GarboWorkshed(
+  $item`Little Geneticist DNA-Splicing Lab`,
+  () => {
+    // This will likely always return true or false for now, depending on the start state of garbo
+    // Since we don't actually support using the syringe in combat at this time, the counter will never change
+    return get("_dnaPotionsMade") >= 3;
+  },
+  () => {
+    // Just grab whatever tonics for now, since we don't actually have support for DNA
+    const page = visitUrl("campground.php?action=workshed");
+    if (page.includes("Human-")) {
+      visitUrl("campground.php?action=dnapotion");
+    }
+  }
+);
+
+const snowMachine = new GarboWorkshed(
+  $item`snow machine`,
+  () => {
+    // Assume garden has already been harvested(?)
+    return true;
+  },
+  () => {
+    return;
+  }
+);
```

Comment:
This and all of the warbear ones that suck feel like they can just be defined nicely with a `.map`